### PR TITLE
Re-add alert confirmation before Cockpit Storage

### PR DESCRIPTION
**Pull request by adamkankovsky**
image

### CODE REVIEW

**Review by KKoukiou — COMMENTED** [review]
Please check tests. :)

**Comment by adamkankovsky** [comment]
> Please check tests. :)

Sure I know about the tests, this was just a checkpoint

**Comment by garrett** [comment]
Nice! It's getting there.

That isn't based on a recent mockup, however. For continence, here's the latest (which I shared in Jira somewhere also, but can't find the specific ticket):

image

The good news is, it's functionally the same, but with a little bit of added text to make it clear that this should not be used in a typical installation.

Text:

> The storage editor lets you resize, delete, and create partitions. It can set up LVM and much more. It is meant to be used as an advanced utility and not intended to be used in a typical installation.
>
> **Any changes you make in the storage editor will immediately affect your devices.**

File: src/components/storage/CockpitStorageIntegration.jsx [hunk 1]
```diff
@@ -130,42 +138,90 @@ export const CockpitStorageIntegration = ({
...
-              isFormDisabled={isFormDisabled}
-              onCritFail={onCritFail}
-              scenarioAvailability={scenarioAvailability}
-              setShowDialog={setShowDialog}
-              setShowStorage={setShowStorage}
-            />}
-        </Modal>
+        <>
+            <Modal
+              title="Modify storage"
```
Comment (KKoukiou) [inline src/components/storage/CockpitStorageIntegration.jsx:156]: Please translate that. _("Modify storage")

File: src/components/storage/CockpitStorageIntegration.jsx [hunk 2]
```diff
@@ -130,42 +138,90 @@ export const CockpitStorageIntegration = ({
...
+              titleIconVariant="warning"
+              aria-describedby="modal-title-icon-description"
```
Comment (KKoukiou) [inline src/components/storage/CockpitStorageIntegration.jsx:158]: Do you need this? You dont even have an icon. I suggest removing it.

File: src/components/storage/CockpitStorageIntegration.jsx [hunk 3]
```diff
@@ -130,42 +138,90 @@ export const CockpitStorageIntegration = ({
...
+                  <Button
+                    id={idPrefix + "-enter-storage-cancel"}
+                    key="cancel"
+                    variant="link"
+                    onClick={handleCancelOpenModal}>
+                      {_("Cancel")}
+                  </Button>
+              ]}
+            >
+                <span id="modal-title-icon-description">
```
Comment (KKoukiou) [inline src/components/storage/CockpitStorageIntegration.jsx:179]: You can make this span wrapper a TextContent.

Then you can use strong text and such stylings by using Text variants.  https://www.patternfly.org/components/text

Then you wont need the <br> i think

**Review by KKoukiou — CHANGES_REQUESTED** [review]
Some small changes needed.

File: src/components/storage/CockpitStorageIntegration.jsx [hunk 4]
```diff
@@ -119,8 +119,16 @@ export const CockpitStorageIntegration = ({
     showStorage,
 }) => {
     const [showDialog, setShowDialog] = useState(false);
+    const [showConfirmation, setShowConfirmation] = useState(false);
+    const [confirmed, setConfirmed] = useState(false);
```
Comment (KKoukiou) [inline src/components/storage/CockpitStorageIntegration.jsx:123]: Using is as a prefix for boolean state variables is a common convention in React and JavaScript coding practices.

For consistency purpose please change this to [isConfirmed, set..]

**Review by KKoukiou — CHANGES_REQUESTED** [review]

File: src/components/storage/CockpitStorageIntegration.jsx [hunk 5]
```diff
@@ -130,42 +138,90 @@ export const CockpitStorageIntegration = ({
...
-              dispatch={dispatch}
-              isFormDisabled={isFormDisabled}
-              onCritFail={onCritFail}
-              scenarioAvailability={scenarioAvailability}
-              setShowDialog={setShowDialog}
-              setShowStorage={setShowStorage}
-            />}
-        </Modal>
+        <>
+            <Modal
```
Comment (KKoukiou) [inline src/components/storage/CockpitStorageIntegration.jsx]: Will be nice to have it as seperate component for readability purposes.

**Review by KKoukiou — COMMENTED** [review]
Did the changes my self so that I can do release including this today.

**Review by KKoukiou — APPROVED** [review]

**Comment by adamkankovsky** [comment]
Thanks for the changes and the suggestions.

**Pull request merged**